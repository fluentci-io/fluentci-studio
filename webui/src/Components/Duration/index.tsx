import Duration from "./Duration";

export default Duration;
