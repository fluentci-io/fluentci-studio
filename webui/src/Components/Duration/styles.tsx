import styled from "@emotion/styled";

export const Container = styled.div`
  font-size: 13px;
  color: #ffffffb8;
  width: 100px;
  text-align: right;
`;
