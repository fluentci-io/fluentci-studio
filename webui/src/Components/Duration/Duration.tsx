import { FC, useEffect, useState } from "react";
import { Container } from "./styles";
import { useFormat } from "../../Hooks/useFormat";
import dayjs from "dayjs";
import durationPlugin from "dayjs/plugin/duration";
import { interval } from "rxjs";

dayjs.extend(durationPlugin);

export type DurationProps = {
  startDate?: string;
  value?: number | null;
};

const Duration: FC<DurationProps> = (props) => {
  const { startDate, value } = props;
  const { formatDuration } = useFormat();
  const [duration, setDuration] = useState<number>();

  useEffect(() => {
    const subscription = interval(1).subscribe(() => {
      if (startDate) {
        setDuration(dayjs().diff(dayjs(startDate), "milliseconds"));
      }
    });
    return () => subscription.unsubscribe();
  }, [startDate]);

  return (
    <>
      {startDate && (
        <Container>
          {duration ? formatDuration(dayjs.duration(duration)) : ""}
        </Container>
      )}
      {!startDate && (
        <Container>
          {value ? formatDuration(dayjs.duration(value)) : ""}
        </Container>
      )}
    </>
  );
};

export default Duration;
