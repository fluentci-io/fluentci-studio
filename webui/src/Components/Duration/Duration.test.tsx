import { render } from "@testing-library/react";
import Duration from "./Duration";

describe("Duration", () => {
  it("should render the component", () => {
    const { container } = render(<Duration value={2000} />);
    expect(container).toMatchSnapshot();
  });
});
