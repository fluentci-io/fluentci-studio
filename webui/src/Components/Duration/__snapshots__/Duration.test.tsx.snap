// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Duration > should render the component 1`] = `
<div>
  <div
    class="css-we0til"
  >
    02.0s
  </div>
</div>
`;
